nexus/reconfigurator/planning/src/blueprint_builder/builder/disks_editor.rs

```diff
@@ -96,14 +96,14 @@ impl BlueprintDisksEditor {
 #[derive(Debug)]
 pub(super) struct SledDisksEditor<'a> {
     config: &'a mut DisksConfig,
-    changed: bool,
+    counts: EditCounts,
     sled_id: SledUuid,
     parent_changed_set: &'a mut BTreeSet<SledUuid>,
 }
 
 impl Drop for SledDisksEditor<'_> {
     fn drop(&mut self) {
-        if self.changed {
+        if self.counts != EditCounts::default() {
```

Comment:
Extremely minor nitpick, but maybe this is worth an `EditCounts::has_changes() -> bool` method?

Reply:
Yeah, seems reasonable. I might go with `has_nonzero_counts()` since this thing itself is just counts, and only the bigger context makes "changes" meaningful.